### PR TITLE
Create with eni

### PR DESCRIPTION
**Pull request by automaticgiant**
here's you a pr @pielu. we still need to get both rebased onto master, then re-check.
sorry for the delay on this.

### CODE REVIEW

**Comment by automaticgiant** [comment]
i got mine rebased, so rebase yours @pielu, maybe fix a merge conflict, and we'll take a look.

File: builtin/providers/aws/resource_aws_instance.go [hunk 1]
```diff
@@ -129,9 +129,11 @@ func resourceAwsInstance() *schema.Resource {
 			},
 
 			"network_interface_id": &schema.Schema{
-				Type:     schema.TypeString,
-				Computed: true,
-				Optional: true,
+
+				Type:          schema.TypeString,
```
Comment (chiradeep) [inline builtin/providers/aws/resource_aws_instance.go:133]: According to http://docs.aws.amazon.com/cli/latest/reference/ec2/run-instances.html, you can create/launch with multiple ENIs, not just 1. So, then, this should be a TypeSet or TypeList?

Comment (automaticgiant) [reply builtin/providers/aws/resource_aws_instance.go:133]: good question. idk anything about TypeSets and TypeLists yet. do you think that the initial implementation should focus on 1 or a list?

Comment (chiradeep) [reply builtin/providers/aws/resource_aws_instance.go:133]: Why not the list? Doesn't look like a whole lot more code. Another question is if the tf config changes this value, should the instance be rebuilt?  That is, should `ForceNew: true`

Comment (pielu) [reply builtin/providers/aws/resource_aws_instance.go:133]: Hi @chiradeep,
Thanks for input.

The problem with the list is that _InstanceNetworkInterfaceSpecification_ requires network interface id **and** index. So Terraform template operator would have to define both at instance creation time and any change would rebuild whole instance (rather undesirable). Additionally it complicates instance resource a bit. That is why I preferred more "atomic" approach, with ability to attach eth0 of given id (versus originally hard-coded new interface) and attaching other enis explicitly (see https://github.com/hashicorp/terraform/pull/10594)

Comment (chiradeep) [reply builtin/providers/aws/resource_aws_instance.go:133]: OK, LGTM. I guess there are not many OS that object to hotplugging NICs after boot

**Pull request merged**